### ISSUE TITLE
ObservationTextHandler class is not defined in a reactive context

### ISSUE THREAD

**Issue by tadlakha9** [spring-projects/spring-security#14651]
**Describe the bug**
While integrating Reactive spring security with observability, we got a code to create the bean of `ObservationRegistryCustomizer` in spring security reference document but in this code, we are trying to call `ObservationTextHandler` but this class is not available.
![image](https://github.com/spring-projects/spring-security/assets/51110539/d60ad6e9-c238-41cb-8302-db89f38e180e)

**To Reproduce**
Steps to reproduce the behavior.

**Expected behavior**
Provide the good sample bean creation while integrating with observability

**Sample**

A link to a GitHub repository with a [minimal, reproducible sample](https://stackoverflow.com/help/minimal-reproducible-example).

Reports that include a sample will take priority over reports that do not.
At times, we may require a sample, so it is good to try and include a sample up front.

**Comment by jzheaux** [comment]
Good catch, @tadlakha9. That appears to be a typo. The class is named `ObservationTextPublisher`. I'll get a fix out with the corrected name.
